src/seedu/addressbook/Main.java

```diff
@@ -109,7 +109,12 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            try {
+                storage.save(addressBook);
+            } catch (StorageOperationException e) {
+                ui.showToUser(e.getMessage());
```

Comment:
The `e.getMessage()` might be the stack trace here. Its possible to specify your own message you want the user to see